time_stretch.py

```diff
@@ -46,20 +42,20 @@ def main(debug=False):
         if debug:
             print(f'  Reading input frame {x}')
 
-        input_frame = Image.open(f'{IN_DIR}/{image_name}')
+        input_frame = Image.open(f'{input_path}/{image_name}')
         for frame_num, output_frame in enumerate(output_images):  # fill that column in each frame
             cropped_image = input_frame.crop((frame_num, 0, frame_num + 1, output_height))
             output_frame.paste(cropped_image, (x, 0))
 
     if debug:
         print('Saving frames')
-    if not os.path.isdir(OUT_DIR):
-        os.mkdir(OUT_DIR)
+    if os.path.exists(output_path):
+        print(f'Deleting directory {output_path}')
+        shutil.rmtree(output_path)
```

Comment:
Can we move this `if` statement to line 12, right after checking the input directory, to follow a fail-fast methodology. Also as I've just suggested in the other file, prefer to fail based on a bad name than to remove existing files.

Reply:
Since these are auto-generated files, I was thinking the user could easily regenerate them if they were deleted accidentally (similarly to how a compiler doesn’t ask before replacing the compiled output file). Specifically, the directory generated by `convert_video_to_frames` seems like an internal detail of the program that the user shouldn’t need to manually delete between runs.

But as I think more about it, I can see scenarios where the user might manually modify the output, so you’re probably right that it’s safer not to delete without asking.